operator/v1/types_ingress.go

```diff
@@ -233,6 +233,97 @@ type LoadBalancerStrategy struct {
 	// +kubebuilder:validation:Required
 	// +required
 	Scope LoadBalancerScope `json:"scope"`
+
+	// providerParameters holds desired load balancer information
+	// specific to the underlying infrastructure provider.
+	//
+	// If empty, defaults will be applied. See specific ProviderLoadBalancerParameters
+	// fields for details about their defaults.
```

Comment:
"ProviderLoadBalancerParameters" does not have meaning in the context of the CRD schema; this will confuse people who are just looking at the CRD or `oc explain` output.  Maybe "ProviderLoadBalancerParameters" → "subfields"?

Reply:
Sorry, the name `providerParameters` was fine; what I meant was that the godoc should not use the Go type in the reference, but rather it should use the CRD field name or a phrase such as "the subfields of this field".  If you look at the yaml definition of the CRD, it doesn't have Go types or Go field names; it only has the text from the godoc and the field names that are specified in the json tags, so the godoc should use those field names, not the Go identifiers.

Reply:
BTW, in hopes I don't cause further confusion, I'll add that godoc for _types_ can use Go identifiers because that godoc does not end up in the CRD definition.  However, godoc for _fields_ should not reference Go identifiers.  Examples:

:arrow_down:  Good, the godoc on the type should have the Go identifier for the type.
```go
// LoadBalancerStrategy holds parameters for a load balancer.
type LoadBalancerStrategy struct {
```
:arrow_down: Good, `providerParameters` should be used in the godoc for the field.
```go
	// providerParameters holds desired load balancer information
	// specific to the underlying infrastructure provider.
	//
	// If empty, defaults will be applied. See specific LoadBalancerSubfields
	// fields for details about their defaults.
	//
	// +optional
	ProviderParameters ProviderLoadBalancerParameters `json:"providerParameters,omitempty"`
```
:arrow_up:  Oops, `LoadBalancerSubfields` is a Go identifier, and this is godoc for a field, so we cannot use Go identifiers here!  `LoadBalancerSubfields` should be changed to `providerParameters` or to some phrase such as "subfields of this field" to avoid having a Go identifier in the CRD definition.

Reply:
@Miciah thanks for the clarification and detailed explanation. I now understand the point you're making.

Reply:
Commit `3e682c0` address this feedback.